packages/cli/src/pages/audiences/index.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { ReactElement } from "react";
+import { NextPage } from "next";
+import { withSessionSsr } from "../../util/session";
+import prisma from "../../../prisma";
+import type { Member, User } from "../../../prisma/generated/client";
+import OutlineButton from "../../components/ui/OutlineButton";
+import Table from "../../components/ui/Table";
+import Link from "next/link";
+import PaginationControl from "../../components/ui/PaginationControl";
+
+const PAGE_SIZE = 20;
+
+type AudiencesProps = {
+  members: Member[];
+  memberListCounts: { [key: string]: number };
+  page: number;
+  total: number;
+  defaultListId: string;
+  user: User;
+  sortOrder: "desc" | "asc";
+};
+
+export const getServerSideProps = withSessionSsr<{ user: any }>(
+  async function ({ req, query }) {
+    const user = req.session.user;
+    const page =
+      "string" === typeof query.page
+        ? Math.max(parseInt(query.page, 10), 1)
+        : 1;
+    const sortOrder = "string" === typeof query.sortOrder ? "asc" : "desc";
+
+    if (!user) {
+      return {
+        redirect: {
+          destination: "/login",
+          permanent: false,
+        },
+      };
+    }
+
+    const membersQuery = prisma.member.findMany({
+      where: {
+        List: {
+          isDefault: true,
+        },
+      },
+      orderBy: {
+        createdAt: sortOrder,
+      },
+      skip: PAGE_SIZE * (page - 1),
+      take: PAGE_SIZE,
+    });
+
+    const totalQuery = prisma.member.count({
+      where: {
+        List: {
+          isDefault: true,
+        },
+      },
+    });
+
+    const [total, members] = await Promise.all([totalQuery, membersQuery]);
+
+    // No members, go back to page 1
+    if (members.length === 0 && page > 1) {
+      return {
+        redirect: {
+          destination: "/audiences",
+          permanent: false,
+        },
+      };
+    }
+
+    let defaultListId: string | undefined = members[0]?.listId;
+    if (!defaultListId) {
+      // no members, so find the default list
+      defaultListId = (
```

Comment:
refactor idea...

defaultListId = (await findOrCreateDefaultList).id

I think we do a similar thing in a bunch of places.